packages/frontend/src/pages/Models.svelte

```diff
@@ -1,39 +1,106 @@
-  import ModelColumnRegistry from './ModelColumnRegistry.svelte';
-  import ModelColumnPopularity from './ModelColumnPopularity.svelte';
-  import ModelColumnLicense from './ModelColumnLicense.svelte';
-  import ModelColumnHw from './ModelColumnHW.svelte';
-
-  const columns: Column<ModelInfo>[] = [
-    new Column<ModelInfo>('Name', { width: '4fr', renderer: ModelColumnName }),
-    new Column<ModelInfo>('HW Compat', { width: '1fr', renderer: ModelColumnHw }),
-    new Column<ModelInfo>('Registry', { width: '1fr', renderer: ModelColumnRegistry }),
-    new Column<ModelInfo>('Popularity', { width: '1fr', renderer: ModelColumnPopularity }),
-    new Column<ModelInfo>('License', { width: '1fr', renderer: ModelColumnLicense }),
-  ];
-  const row = new Row<ModelInfo>({});
+import type { ModelInfo } from '@shared/models/IModelInfo';
+import NavPage from '../lib/NavPage.svelte';
+import Table from '../lib/table/Table.svelte';
+import { Column, Row } from '../lib/table/table';
+import { localModels } from '../stores/local-models';
+import ModelColumnName from './ModelColumnName.svelte';
+import ModelColumnRegistry from './ModelColumnRegistry.svelte';
+import ModelColumnPopularity from './ModelColumnPopularity.svelte';
+import ModelColumnLicense from './ModelColumnLicense.svelte';
+import ModelColumnHw from './ModelColumnHW.svelte';
+import { onDestroy, onMount } from 'svelte';
+import { studioClient } from '/@/utils/client';
+import type { Category } from '@shared/models/ICategory';
+import type { Task } from '@shared/models/ITask';
+import TasksProgress from '/@/lib/progress/TasksProgress.svelte';
+import { faRefresh } from '@fortawesome/free-solid-svg-icons';
+import Card from '/@/lib/Card.svelte';
+import Button from '/@/lib/button/Button.svelte';
+import LinearProgress from '/@/lib/progress/LinearProgress.svelte';
+
+const columns: Column<ModelInfo>[] = [
+  new Column<ModelInfo>('Name', { width: '4fr', renderer: ModelColumnName }),
+  new Column<ModelInfo>('HW Compat', { width: '1fr', renderer: ModelColumnHw }),
+  new Column<ModelInfo>('Registry', { width: '1fr', renderer: ModelColumnRegistry }),
+  new Column<ModelInfo>('Popularity', { width: '1fr', renderer: ModelColumnPopularity }),
+  new Column<ModelInfo>('License', { width: '1fr', renderer: ModelColumnLicense }),
+];
+const row = new Row<ModelInfo>({});
+
+let loading: boolean = true;
+let intervalId: ReturnType<typeof setInterval> | undefined = undefined;
+
+$: tasks = [] as Task[];
+$: models = [] as ModelInfo[];
+$: filteredModels = [] as ModelInfo[];
+
+function filterModels(): void {
+  // Let's collect the models we do not want to show (loading, error).
+  const modelsId: string[] = tasks.reduce((previousValue, currentValue) => {
+    if(currentValue.state === 'success')
+      return previousValue;
+
+    if(currentValue.labels !== undefined) {
+      previousValue.push(currentValue.labels["model-pulling"]);
+    }
+    return previousValue;
+  }, [] as string[]);
```

Comment:
I feel it could be easier to read/maintain with a `tasks.filter(...).map(t => t.labels['model-pulling']`

Reply:
I agree on the reading level, but not on the performance side as we are iterating two time instead of one, as this method can be called a lot of time I would prefer adding some comments to improve the readability